playground/backend/cmd/server/tcp.go

```diff
@@ -0,0 +1,44 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"beam.apache.org/playground/backend/internal/environment"
+	"beam.apache.org/playground/backend/internal/logger"
+	"context"
+	"google.golang.org/grpc"
+	"net"
+)
+
+// listenTcp binds the grpc.Server to the TCP address
+func listenTcp(ctx context.Context, errChan chan error, envs environment.NetworkEnvs, svc *grpc.Server) {
+	logger.Infof("listening TCP at %s\n", envs.Address())
+	lis, err := net.Listen("tcp", envs.Address())
+	if err != nil {
+		errChan <- err
+		return
+	}
+	if err := svc.Serve(lis); err != nil {
+		errChan <- err
+		return
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		}
```

Comment:
Just to improve my understanding - what does this do?

Reply:
Pablo, it means when the cancel method of goroutine context is called, the case <- ctx.Done will be triggered in  select statement and current goroutine will be stopped.